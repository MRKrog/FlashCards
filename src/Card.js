import React, { Component } from 'react';
import BtnAnswer from './BtnAnswer.js'

import './styles/Card.scss';

class Card extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const answerArray = this.props.currentCard.answers;
    const cardQuestion = this.props.currentCard.question;

    const cardAnswers = answerArray.map((answer) =>
        <BtnAnswer key={answer}
                   answer={answer}
                   onChangeCard={this.props.onChangeCard}
        />
    );

    return (
      <article className="Card">
        <section className="Card-question">
          <h3>{cardQuestion}</h3>
        </section>
        <section className="Card-answers">
          {cardAnswers}
        </section>
      </article>
    )
  }
}


export default Card;
