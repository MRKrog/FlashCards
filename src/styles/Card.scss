$font-title: 'Abel', sans-serif;
$font-header: 'Rajdhani', sans-serif;
$font-copy: 'Lato', sans-serif;

$white: #fff;
$black: #231f20;
$dark-blue: #11212a;

.Card {
  background: $black;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.8);
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 2em;
  width: 100%;
  max-width: 960px;

  .Card-question {

    h3 {
      margin: 0;
      font-family: $font-title;
      color: #fff;
      font-size: 2em;
      line-height: 1;
      padding-bottom: 1em;
      font-weight: 100;
    }
  }

  .Card-answers {
    width: 100%;
  }

}
