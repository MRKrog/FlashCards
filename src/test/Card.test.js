import React from "react";
import MockData from "../mockData";
import Card from "../Card";
import { shallow } from "enzyme";

const currentCard = {
      "id": 101,
      "category": "html",
      "question": "What does HTML stand for?",
      "correctAnswer": "Hyper Text Markup Language",
      "answers": [
      "Home Tool Markup Language",
      "Hyperlinks and Text Markup Language",
      "Hyper Text Material Language",
      "Hyper Text Markup Language"]
    }

const answers = currentCard.answers
const question = currentCard.question;
const onChangeCard = jest.fn();

describe("Card", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Card key={currentCard.id}
            currentCard={currentCard}
            onChangeCard={onChangeCard}
            />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

});
