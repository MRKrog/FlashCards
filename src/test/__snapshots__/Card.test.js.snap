// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot when all data is passed correctly 1`] = `
<article
  className="Card"
>
  <section
    className="Card-question"
  >
    <h3>
      What does HTML stand for?
    </h3>
  </section>
  <section
    className="Card-answers"
  >
    <BtnAnswer
      answer="Home Tool Markup Language"
      key="Home Tool Markup Language"
      onChangeCard={[MockFunction]}
    />
    <BtnAnswer
      answer="Hyperlinks and Text Markup Language"
      key="Hyperlinks and Text Markup Language"
      onChangeCard={[MockFunction]}
    />
    <BtnAnswer
      answer="Hyper Text Material Language"
      key="Hyper Text Material Language"
      onChangeCard={[MockFunction]}
    />
    <BtnAnswer
      answer="Hyper Text Markup Language"
      key="Hyper Text Markup Language"
      onChangeCard={[MockFunction]}
    />
  </section>
</article>
`;
